src/ddmd/hdrgen.d

```diff
@@ -574,13 +574,11 @@ public:
         buf.writenl();
         buf.writestring("finally");
         buf.writenl();
-        buf.writeByte('{');
-        buf.writenl();
-        buf.level++;
+        if (!s.finalbody.isScopeStatement())
+            buf.level++;
         s.finalbody.accept(this);
-        buf.level--;
-        buf.writeByte('}');
-        buf.writenl();
+        if (!s.finalbody.isScopeStatement())
+            buf.level--;
```

Comment:
This looks better, reduces the risk of copy/pasta errors, and does not use negation:
```
if (s.finalbody.isScopeStatement())
    s.finalbody.accept(this);
else
{
    ++buf.level;
    s.finalbody.accept(this);
    --buf.level;
}
```
(It's good practice to prefer code arrangements that avoid logical negation, as humans have cognitive problems with negation.)

Also, this needs a test case for the red statements. If it is not possible to execute those statements, then the code should be removed and replaced with an `assert`.